Shop/Product.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Shop
+{
+    public class Product
+    {
+        public int Code { get; set; }
+
+        public string ProductName { get; set; }
+
+        public int Quantity { get; set; }
+
+        public int Price { get; set; }
+
+        public Product(int code, string name, int quantity, int price)
+        {
```

Comment:
It is good practice to validate parameters for incorrect values for all public methods/ctors.